tests/test_represent.py

```
@@ -18,6 +19,21 @@ def test_standard_represent():
     logger.info("✅ test standard represent function done")
 
 
+def test_standard_represent_with_io_object():
+    img_path = "dataset/img1.jpg"
+    defualt_embedding_objs = DeepFace.represent(img_path)
+    io_embedding_objs = DeepFace.represent(open(img_path, 'rb'))
```

what if i pass a text file as

```
io_obj = io.BytesIO(open("requirements.txt", 'rb').read())
```

---

Yep, that'll work as `io.BytesIO` supports `.read`.

---

this is not okay then, we should be able to pass only image files to deepface functionalities.

---

Oh, I'm sorry, I misunderstood what you were asking.

An error would be raised by `np.frombuffer(obj.read(), np.uint8)` or `cv2.imdecode(nparr, cv2.IMREAD_COLOR)`, just as if you were trying to pass any other non-supported filetype as a string/filepath.


I only meant it would pass the `hasattr(img, "read") and callable(img.read)`

---

I tried to call `load_image_from_io_object` with requirements.txt but it didn't throw any exception

---

I see. I had been following how `load_image_from_web` handled malformed images, but, if you want to raise the error in the function, I could update `load_from_io_object` to raise a ValueError if `cv2.imdecode` returns `None` like how `load_image_from_file_storage` is currently implemented.

It does appear modules in `deepface.modules` all already do `None` checks when loading the `img_path`, which `load_image_from_io_object` does follow in convention.

---

Would you please raise an error if it is not an image?

---

`load_image_from_io_object` now raises a `ValueException` if `cv2.imdecode` returns `None` for objects that aren't images, which is in line with how `load_image_from_file_storage` handles non-image objects.